examples/example_2/makefile

```diff
@@ -5,31 +5,31 @@
 # ==========================================
 
 #We try to detect the OS we are running on, and adjust commands as needed
-ifeq ($(OSTYPE),cygwin)
-	CLEANUP = rm -f
-	MKDIR = mkdir -p
-	TARGET_EXTENSION=.out
-elseifeq ($(OSTYPE),msys)
-	CLEANUP = rm -f
-	MKDIR = mkdir -p
-	TARGET_EXTENSION=.exe
-elseifeq ($(OS),Windows_NT)
+ifeq ($(OS),Windows_NT)
+  ifeq ($(shell uname -s),) # not in a bash-like shell
 	CLEANUP = del /F /Q
 	MKDIR = mkdir
+  else # in a bash-like shell, like msys
+	CLEANUP = rm -f
+	MKDIR = mkdir -p
+  endif
 	TARGET_EXTENSION=.exe
 else
 	CLEANUP = rm -f
 	MKDIR = mkdir -p
 	TARGET_EXTENSION=.out
 endif
 
-UNITY_ROOT=../..
 C_COMPILER=gcc
+ifeq ($(shell uname -s), Darwin)
+C_COMPILER=clang
+endif
+
+UNITY_ROOT=../..
 
-CFLAGS = -std=c99
+CFLAGS=-std=c99
 CFLAGS += -Wall
 CFLAGS += -Wextra
-CFLAGS += -Werror 
```

Comment:
The examples compile without any warnings on most versions of gcc, nor did they throw warnings before. I also removed a flag that was for C++, and checked the two examples have the same warning flags.

I removed `-Werror` specifically due to how warnings change in compilers over time. I recently found out that gcc 6.1 can issue a warning for inconsistent & misleading indentation. It's even on by default. This applies here, since the example uses misleading indentation in the code. (An example of unit testing catching mistakes)
If `-Werror` is enabled, up-to-date users will download the example and find it doesn't build at all.
Philosophically, this compromise felt better for users, to me at least.
I wasn't ready to change up the whole example. Thoughts?

Also, this branch is open to edits from maintainers. You can commit to it and make changes, if you'd like.